sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -1,48 +1,140 @@
+SO_SEQ_FEAT_ROLE_NS = "http://identifiers.org/so/"
+GB2SO_MAPPINGS_CSV = os.path.join(os.path.dirname(os.path.realpath(__file__)), "gb2so.csv")
+SO2GB_MAPPINGS_CSV = os.path.join(os.path.dirname(os.path.realpath(__file__)), "so2gb.csv")
+
+
+class GenBank_SBOL3_Converter:
+    gb2so_map = {}
+    so2gb_map = {}
+    default_SO_ontology = "SO:0000110"
+    default_GB_ontology = "" # TODO: Whats the default here?
+    def create_GB_SO_role_mappings(self, gb2so_csv: str = GB2SO_MAPPINGS_CSV, so2gb_csv: str = SO2GB_MAPPINGS_CSV, 
+                                   convert_gb2so: bool = True, convert_so2gb: bool = True):
+        """Reads 2 CSV Files containing mappings for converting between GenBank and SO ontologies roles
+        :param gb2so_csv: path to read genbank to so conversion csv file
+        :param so2gb_csv: path to read so to genbank conversion csv file
+        :param convert_gb2so: bool stating whether to read csv for gb2so mappings
+        :param convert_so2gb: bool stating whether to read csv for so2gb mappings
+        """
+        if convert_gb2so:
+            logging.info("Reading GB to SO ontology mappings csv")
+            with open(gb2so_csv, mode='r') as csv_file:
+                csv_reader = csv.DictReader(csv_file)
+                for row in csv_reader:
+                    self.gb2so_map[row["GenBank_Ontology"]] = row["SO_Ontology"]
+        if convert_so2gb:
+            logging.info("Reading SO to GB ontology mappings csv")
+            with open(so2gb_csv, mode='r') as csv_file:
+                csv_reader = csv.DictReader(csv_file)
+                for row in csv_reader:
+                    self.so2gb_map[row["SO_Ontology"]] = row["GenBank_Ontology"]
+
+
+    def convert_genbank_to_sbol3(self, gb_file: str, sbol3_file: str = "sbol3.out", namespace: str =
+                                TEST_NAMESPACE, write: bool = False) -> sbol3.Document:
+        """Convert a GenBank document on disk into an SBOL3 document
+        Specifically, the GenBank document is first imported to SBOL2, then converted from SBOL2 to SBOL3
+
+        :param gb_file: path to read GenBank file from
+        :param sbol3_file: path to write SBOL3 file to, if write set to true
+        :param namespace: URIs of Components will be set to {namespace}/{genbank_id},
+                          defaults to "https://test.sbol3.genbank/" 
+        :param write: writes the generated sbol3 document in SORTED_NTRIPLES
+                          format to provided sbol3_file path
+        :return: SBOL3 document containing converted materials
+        """
+        # create sbol3 document, and record parser handler for gb file
+        sbol3.set_namespace(namespace)
+        doc = sbol3.Document()
+        # access records by parsing gb file using SeqIO class
+        logging.info("Parsing Genbank records using SeqIO class.")
+        records = list(SeqIO.parse(gb_file, "genbank").records)
```

Comment:
There's probably no need to read the entire genbank file into memory like this. You can instead iterate over the records one at a time a few lines down with:

```python
    for record in SeqIO.parse(gb_file, "genbank"):
```

Reply:
I see, I have edited this for both records and features in the latest commit.